go-app-ussd_pmtct.js

```diff
@@ -0,0 +1,267 @@
+                text: questions[name],
+                next: "state_start"
+            });
+        });
+
+        self.add("state_birth_year", function(name) {
+            return new FreeText(name, {
+                question: questions[name],
+                /*check: function(content) {
+                    return (utils.check_valid_number(content)
+                        & utils.check_number_in_range(content, "1900", utils.get_today().year)
+                },*/
+                next: function(content) {
+                    self.im.user.set_answer("dob_year", content);
+                    return "state_birth_month";
+                }
+            });
+        });
+
+        self.add("state_birth_month", function(name) {
+            return new ChoiceState(name, {
+                question: questions[name],
+                choices: [
+                    new Choice("jan", $("Jan")),
+                    new Choice("feb", $("Feb")),
+                    new Choice("mar", $("Mar")),
+                    new Choice("apr", $("Apr")),
+                    new Choice("may", $("May")),
+                    new Choice("jun", $("Jun")),
+                    new Choice("jul", $("Jul")),
+                    new Choice("aug", $("Aug")),
+                    new Choice("sep", $("Sep")),
+                    new Choice("oct", $("Oct")),
+                    new Choice("nov", $("Nov")),
+                    new Choice("dec", $("Dec"))
+                ],
+                next: function(content) {
+                    self.im.user.set_answer("dob_month", content);
+                    return "state_birth_day";
+                }
+            });
+        });
+
+        self.add("state_birth_day", function(name) {
+            return new FreeText(name, {
+                question: questions[name],
+                /*check: function(content) {
+                    utils.is_valid_date(dob)  // check here or in "next"
+                },*/
+                next: function(content) {
+                    self.im.user.set_answer("dob_day", content);
+                    /*self.im.user.set_answer("dob",
+                        utils.get_entered_birth_date(self.im.user.dob_year, self.im.user.dob_month, content));*/
+
+                    return "state_hiv_messages";
+                }
+            });
+        });
+
+        self.add("state_hiv_messages", function(name) {
+            return new ChoiceState(name, {
+                question: questions[name],
+                // error: ,
+                choices: [
+                    new Choice("yes", $("Yes")),
+                    new Choice("no", $("No"))
+                ],
+                next: function(choice) {
+                    if (choice.value === "yes") {
+                        self.im.user.set_answer("hiv_opt-in", "true");
+                        return "state_end_hiv_messages_confirm";
+                    } else {
+                        self.im.user.set_answer("hiv_opt-in", "false");
```

Comment:
I'm not sure if the dash will cause problems at some point - it probably won't but I like to just stick with underscores.